airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -144,7 +133,10 @@ def stream_slices(self, sync_mode: SyncMode, stream_state: Mapping[str, Any]) ->
         stream_state = stream_state or {}
         kwargs = {"stream_state": stream_state}
         end_datetime = min(self.end_datetime.get_datetime(self.config, **kwargs), datetime.datetime.now(tz=self._timezone))
-        lookback_delta = self._parse_timedelta(self.lookback_window.eval(self.config, **kwargs) if self.lookback_window else "0d")
+        if self.lookback_window and (evaluated_lookback_window := self.lookback_window.eval(self.config, **kwargs)):
+            lookback_delta = self._parse_timedelta(evaluated_lookback_window)
+        else:
+            lookback_delta = self._parse_timedelta("P0D")
```

Comment:
Is this still what we want if user input for granularity is > 1 day? Wondering if that could result in scenario 1 (from your PR description) issues again.

Reply:
I changed this code because the previous `_parse_timedelta` was taking an empty string as a parameter which makes me think I should have keep the behavior it had before in case someone else calls it with an empty string so I'll fix that.

I think this is still what we want. If a user specifies a lookback, I think we should use it whatever it is. Maybe we could have an error if the lookback_window is smaller than the granularity since the impact would be that the lookback_window applied would be the granularity and not the lookback per se. For example:
* lookback_window = 1 second
* datetimeformat = "%y-%m-%d"
* granularity = 1 day
* start_datetime = 2022-12-15T00:00:00Z

In this case, we would `2022-12-15T00:00:00Z - 1 second` which would give `2022-12-14T23:59:00Z`. However, when formatting this, it ends up being `2022-12-14` and therefore, the lookback window is 1 day, not 1 second. @girarda can you confirm a breaking exception here with a nice error message would make sense? Are there side effects that we would not want and maybe just logging is better?